subsys/net/lib/config/init_clock_sntp.c

```diff
@@ -112,10 +112,230 @@ int net_init_clock_via_sntp(void)
 }
 
 #ifdef CONFIG_NET_CONFIG_SNTP_INIT_RESYNC
+static void sntp_async_service_handler(struct net_socket_service_event *pev);
+NET_SOCKET_SERVICE_SYNC_DEFINE_STATIC(service_sntp_async, sntp_async_service_handler, 1);
+
+static void sntp_async_timeout(struct k_work *work);
+static K_WORK_DELAYABLE_DEFINE(sntp_async_timeout_work, sntp_async_timeout);
+
+static struct sntp_ctx sntp_async_ctx;
+static struct net_sockaddr_storage sntp_addr;
+static net_socklen_t sntp_addrlen;
+
+#define RESYNC_FAILED_INTERVAL K_SECONDS(CONFIG_NET_CONFIG_SNTP_INIT_RESYNC_ON_FAILURE_INTERVAL)
+
+static void sntp_async_timeout(struct k_work *work)
+{
+	ARG_UNUSED(work);
+
+	LOG_DBG("SNTP query timed out");
+
+	sntp_close_async(&service_sntp_async);
+
+	/* No response, reschedule */
+	k_work_reschedule(&sntp_resync_work_handle, RESYNC_FAILED_INTERVAL);
+}
+
+static void sntp_async_service_handler(struct net_socket_service_event *pev)
+{
+	struct timespec tspec;
+	struct sntp_time ts;
+	int ret;
+
+	ret = sntp_read_async(pev, &ts);
+	if (ret < 0) {
+		LOG_ERR("Failed to read SNTP response (%d)", ret);
+		goto out;
+	}
+
+	tspec.tv_sec = ts.seconds;
```

Comment:
These lines are very identical to `net_init_clock_via_sntp`. Can you wrap them in a function?

Reply:
Added helper function for setting clocs.